presto-main/src/main/java/com/facebook/presto/sql/planner/Partitioning.java

```diff
@@ -168,6 +172,16 @@ public boolean isRefinedPartitioningOver(
         return true;
     }
 
+    public IdentityHashMap<RowExpression, RowExpression> mapPartitioningArguments(Partitioning other)
+    {
+        checkArgument(getArguments().size() == other.getArguments().size());
+        IdentityHashMap<RowExpression, RowExpression> argumentsMap = new IdentityHashMap<>();
```

Comment:
I think the assumption that two partitioning cannot have an exactly the same `RowExpression` object may not always hold. How about creating an inner class, e.g.: `PartitioningArgumentMapping` containing two fields, `RowExpression first`, `RowExpression second`.

Then, instead of translating the `probeProperties` with the `rewriteToColeasce` you can simply iterate the mapings, and create coleasce expressions based on them (you can add assertions needed to make sure the partitioning arguments match the equi join condition).

Reply:
I think even when they have the same value in `RowExpression` they should still be two distinct objects right? Or you think it's possible that they are exactly the same object?

Reply:
I prefer to use translator if possible, so it's the same as how argument bindings are processed at other places.